### PR TITLE
Properly identify the originating IP address of a client connecting through HTTP proxy or load balancer

### PR DESCRIPTION
**Pull request by yspro**
Properly identify the originating IP address of a client connecting through HTTP proxy or load balancer

### CODE REVIEW

File: lib/kiev/rack/request_logger.rb [hunk 1]
```diff
@@ -70,6 +70,11 @@ def log_rack_exception?(exception)
         !Config.instance.ignored_rack_exceptions.include?(exception.class.name)
       end
 
+      def real_ip(request)
+        # the Rack::Request#ip is broken for ages
```
Comment (Raffo) [inline lib/kiev/rack/request_logger.rb]: nit: I would drop this comment and add an explanation on this PR on why this is broken for ages.

Comment (lipanski) [reply lib/kiev/rack/request_logger.rb]: not sure if `ActionDispatch` is always there - I've seen things like `if defined?(ActionDispatch)` in here (e.g. `lib/kiev/param_filter.rb`).

File: lib/kiev/rack/request_logger.rb [hunk 2]
```diff
@@ -70,6 +70,11 @@ def log_rack_exception?(exception)
...
+        ActionDispatch::Request.new(request.env).remote_ip
```
Comment (lipanski) [inline lib/kiev/rack/request_logger.rb]: not sure if `ActionDispatch` is always there - I've seen things like `if defined?(ActionDispatch)` in here (e.g. `lib/kiev/param_filter.rb`).

File: lib/kiev/rack/request_logger.rb [hunk 3]
```diff
@@ -70,6 +70,15 @@ def log_rack_exception?(exception)
         !Config.instance.ignored_rack_exceptions.include?(exception.class.name)
       end
 
+      def real_ip(request)
+        # use X_FORWARDED_FOR when request comes from load_balancer/proxy
+        if request.env["X_FORWARDED_FOR"]
```
Comment (lipanski) [inline lib/kiev/rack/request_logger.rb]: `X-Forwarded-For` is (or can be) a list of IPs = all the proxy servers the request has passed through.

so I'd rather go with this approach: https://github.com/rack/rack/blob/master/lib/rack/request.rb#L270-L273

(including rejecting the trusted IPs = local networks etc.)

Comment (yspro) [reply lib/kiev/rack/request_logger.rb]: @lipanski  you're totally right and I didn't change that behavior, the `Rack::Request#ip` is still used, see [here](https://github.com/blacklane/kiev/blob/feature/real-ip/lib/kiev/rack/request_logger.rb#L18)

I'm just replacing `REMOTE_ADDR` with `X-Forwarded-For`, look at the new test I've just added https://github.com/blacklane/kiev/pull/6/commits/b759425e5a120bc273f0f8885f998b45b4672530

**Comment by lipanski** [comment]
I guess we can close this now? @yspro

**Comment by yspro** [comment]
@lipanski yes

**Pull request closed without merging**